src/prelude.rs

```diff
@@ -0,0 +1,8 @@
+//! Crate-local prelude (for alloc-dependent features like `Vec`)
+
+// TODO: switch to alloc::prelude
```

Comment:
I can go ahead and do this now if you're happy with requiring recent nightlies

Reply:
I'm happy with merging this PR without that and adding it later.

Reply:
Sounds good. The only thing you're using is `Vec` anyway, so no need to pull in the whole prelude.